### DISCUSSION TITLE
Keep file sources in sync with created tables

### DISCUSSION THREAD

**Discussion by alejandrofm** [duckdb/duckdb#8660] [Ideas]
Hi, it will be nice if:
- Table was created like this:
`CREATE TABLE t1 AS SELECT * FROM  read_parquet('/tmp/000.parquet')`

- And a change has been made to the source file(s)
DuckDb fires a refresh of the table.

But I also want to know how you keep your tables up to date when a source file can change 1-100 times throughout the day, considering that the duckdb file format is better than parquet. And maintaining the table name.

Use case: DB backend for an analytics tool connected via SQLAlchemy.
thanks!

**Comment by l1t1** [comment]
you can use a view

**Comment by l1t1** [comment]
```sql
D copy (select 42 a ) to 'd:/42a.parquet';
D create view v as select * from 'd:/42a.parquet';
D select * from v;
┌───────┐
│   a   │
│ int32 │
├───────┤
│    42 │
└───────┘
D copy (select 42 a union select 43) to 'd:/42a.parquet';
D select * from v;
┌───────┐
│   a   │
│ int32 │
├───────┤
│    43 │
│    42 │
└───────┘
```

**Reply by alejandrofm** [reply]
Thanks! @l1t1 To optimize reads, can I create a view from partitioned parquet files?
Didn't find in the documentation how to tell the view how the file is partitioned (in my case the column is data_date)

CREATE VIEW v as SELECT * FROM read_parquet('test/*/*.parquet', hive_partitioning=true);

Also, I infer that in this case, I cannot use enable_object_cache either because of the volatility of the source, or duckdb can tell if there was a difference/set a TTL to the cache?

Looking at how can I get the most performance from duckdb while using only views.

The second option I'm thinking about are tables+views+DBT to keep the tables updated.

Thanks a lot!
